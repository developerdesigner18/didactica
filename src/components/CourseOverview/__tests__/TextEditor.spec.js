import { waitRAF } from "@/../tests/utils.js";
import { createLocalVue, mount } from "@vue/test-utils";
import LxpTextEditor from "../TextEditor.vue";

const localVue = createLocalVue();

describe("TextEditor.vue", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = await getWrapper();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it("renders when created", async () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  describe("Props", () => {
    it("renders when created", async () => {
      const wrapper = mount(LxpTextEditor);
      expect(wrapper.exists()).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
      wrapper.destroy();
    });
    it("should update id based on prop", async () => {
      await wrapper.setProps({ id: "new" });
      await waitRAF();
      expect(wrapper.find("#new").exists()).toBe(true);
    });

    it("should display support text when provided", async () => {
      const text = "Support text here";
      await wrapper.setData({ cSupportText: text });
      await waitRAF();
      expect(wrapper.find(".lxp-editor__footer--support-text").text()).toBe(
        text
      );
    });
  });
});

function getWrapper() {
  return mount(LxpTextEditor, {
    propsData: {
      value: "Initial content",
      placeholder: "Type something in here!",
      limit: 100,
      hasError: false,
      supportText: "Support text here"
    },
    localVue
  });
}
