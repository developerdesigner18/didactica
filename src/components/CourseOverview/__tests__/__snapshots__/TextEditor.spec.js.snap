// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextEditor.vue Props renders when created 1`] = `
<div
  class="lxp-text_editor"
>
  <!---->
   
  <div
    class="quill-editor ql-container ql-bubble"
    id="htmlEditor"
  >
    <div
      class="ql-editor ql-blank"
      contenteditable="true"
      data-gramm="false"
      data-placeholder="Type something in here!"
    />
    <div
      class="ql-clipboard"
      contenteditable="true"
      tabindex="-1"
    />
    <div
      class="ql-tooltip ql-hidden"
    >
      <span
        class="ql-tooltip-arrow"
      />
      <div
        class="ql-tooltip-editor"
      >
        <input
          data-formula="e=mc^2"
          data-link="  Paste link here and press enter"
          data-video="Embed URL"
          type="text"
        />
        <a
          class="ql-close"
        />
      </div>
      <div
        class="ql-toolbar"
      >
        <span
          class="ql-formats"
        >
          <button
            class="ql-bold"
            type="button"
          />
          <button
            class="ql-italic"
            type="button"
          />
        </span>
        <span
          class="ql-formats"
        >
          <button
            class="ql-list"
            type="button"
            value="ordered"
          />
          <button
            class="ql-list"
            type="button"
            value="bullet"
          />
        </span>
        <span
          class="ql-formats"
        >
          <button
            class="ql-link"
            type="button"
          />
        </span>
      </div>
    </div>
  </div>
   
  <div
    class="lxp-text_editor__footer"
  >
    <!---->
     
    <div
      class="lxp-editor__footer--limit"
    >
      
      0/100
    
    </div>
  </div>
</div>
`;
